src/__test__/App.test.js

```diff
@@ -0,0 +1,23 @@
+import React from 'react';
+import { render } from '@testing-library/react';
+import { BrowserRouter as Router } from 'react-router-dom';
+import { Provider } from 'react-redux';
+/* eslint-disable */
+import store from '../redux/store';
+import App from '../App';
+/* eslint-enable */
+
+describe('Coins test', () => {
+  test('snapshot for coins', () => {
+    const tree = render(
+      <React.StrictMode>
+        <Provider store={store}>
+          <Router>
+            <App />
+          </Router>
+        </Provider>
+      </React.StrictMode>,
+    );
+    expect(tree).toMatchSnapshot();
+  });
+});
```

Comment:
- [ ] Kindly make sure that the App has also integration tests using **React Testing Library** as required for the project. Right now you only do unit tests for some components.  You can perform a mocking data fetching and Redux Store connection. For more detail about **integrations tests**, I suggest you **[watch this short video](https://www.youtube.com/watch?v=6wbnwsKrnYU)**.